receiver/k8sobjectsreceiver/unstructured_to_logdata.go

```diff
@@ -0,0 +1,73 @@
+// Copyright  The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package k8sobjectsreceiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/k8sobjectsreceiver"
+
+import (
+	"go.opentelemetry.io/collector/pdata/plog"
+	semconv "go.opentelemetry.io/collector/semconv/v1.9.0"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/watch"
+)
+
+func watchEventToLogData(event watch.Event) plog.Logs {
+	udata := event.Object.(*unstructured.Unstructured)
+	out := plog.NewLogs()
+	rl := out.ResourceLogs().AppendEmpty()
+	sl := rl.ScopeLogs().AppendEmpty()
+	lr := sl.LogRecords().AppendEmpty()
+	dest := lr.Body()
+
+	attrs := lr.Attributes()
+	attrs.EnsureCapacity(3)
+
+	attrs.PutString("event.domain", "k8s")
+	attrs.PutString("event.name", udata.GetKind())
+
+	if namespace := udata.GetNamespace(); namespace != "" {
+		attrs.PutString(semconv.AttributeK8SNamespaceName, namespace)
+	}
+
+	destMap := dest.SetEmptyMap()
+	obj := map[string]interface{}{
+		"type":   string(event.Type),
+		"object": udata.Object,
```

Comment:
Why do we wrap it in this function but put Object into the log body as is in unstructuredListToLogData?

Reply:
When using watch mode, Kubernetes will return object as following structure:
```json
{
    "type": "ADDED",
    "object": {
        "kind": "Event",
        "apiVersion": "v1",
        "metadata": {
            "name": "counter.150b2bf03f8963dd",
            "namespace": "default",
            "selfLink": "/api/v1/namespaces/default/events/counter.150b2bf03f8963dd",
            "uid": "e0effe55-fcfa-11e7-95cd-000d3af99127",
            "resourceVersion": "6621683",
            "creationTimestamp": "2018-01-19T09:26:50Z"
        },
        "involvedObject": {
            "kind": "Pod",
            "namespace": "default",
            "name": "counter",
            "uid": "dec534aa-fcfa-11e7-95cd-000d3af99127",
            "apiVersion": "v1",
            "resourceVersion": "6621674",
            "fieldPath": "spec.containers{count}"
        },
        "reason": "Created",
        "message": "Created container",
        "source": {
            "component": "kubelet",
            "host": "k8s-agentpool1-10851795-1"
        },
        "firstTimestamp": "2018-01-19T09:26:50Z",
        "lastTimestamp": "2018-01-19T09:26:50Z",
        "count": 1,
        "type": "Normal"
    }
}
```
I have wrapped `object` to retain the original structure

Reply:
Can we reduce the code duplication between the functions? Maybe use something like the following instead `unstructuredListToLogData`

```go
eventsToLogData(events unstructured.UnstructuredList{
	Type: string(event.Type),
	Items: []unstructured.Unstructured{{Object: udata.Object}}
}
```

Reply:
Done